common/stats/stats.go

```diff
@@ -0,0 +1,479 @@
+type StatsRegistry interface {
+	// Gets an existing metric or registers the given one.
+	// The interface can be the metric to register if not found in registry,
+	// or a function returning the metric for lazy instantiation.
+	GetOrRegister(string, interface{}) interface{}
+
+	// Unregister the metric with the given name.
+	Unregister(string)
+
+	// Call the given function for each registered metric.
+	Each(func(string, interface{}))
+}
+
+//
+// A registry wrapper for metrics that will be collected about the runtime
+// performance of an application.
+//
+// A quick note about name elements: hierarchical names are stored using a '/'
+// path separator. To avoid confusion, variadic name elements passed to any
+// method will have '/' characters in their names replaced by the string
+// "_SLASH_" before they are used internally. This is instead of failing,
+// because sometimes counters are dynamically generated (i.e. with error
+// names), and it is better to strip the path elements than to, for example,
+// panic.
+//
+type StatsReceiver interface {
+	// Return a stats receiver that will automatically namespace elements with
+	// the given scope args.
+	//
+	//   statsReceiver.Scope("foo", "bar").Stat("baz")  // is equivalent to
+	//   statsReceiver.Stat("foo", "bar", "baz")
+	//
+	Scope(scope ...string) StatsReceiver
+
+	// If StatsRegistry supports the latency instrument:
+	//
+	// Returns a copy that can in turn create a Latency instrument that will use the
+	// given precision as its display precision when the stats are rendered as
+	// JSON. For example:
+	//
+	//   statsReceiver.Precision(time.Millisecond).Stat("foo_ms")
+	//
+	// means that the 'foo_ms' stat will have its nanosecond data points displayed
+	// as milliseconds when rendered. Note that this does _not_ affect the
+	// captured data in any way, only its display.
+	//
+	// If the given duration is <= 1ns, we will default to ns.
+	Precision(time.Duration) StatsReceiver
+
+	// Provides an event counter
+	Counter(name ...string) Counter
+
+	// Provides a histogram of sampled stats over time. Times output in
+	// nanoseconds by default, but can be adjusted by using the Precision()
+	// function.
+	Latency(name ...string) Latency
+
+	// Add a gauge, which holds an int64 value that can be set arbitrarily.
+	Gauge(name ...string) Gauge
+
+	// Add a gauge, which holds a float64 value that can be set arbitrarily.
+	GaugeFloat(name ...string) GaugeFloat
+
+	// Removes the given named stats item if it exists
+	Remove(name ...string)
+
+	// Construct a JSON string by marshaling the registry.
+	Render(pretty bool) []byte
+}
+
+//
+// DefaultStats is a small wrapper around a go-metrics like registry.
+// Uses defaultStatsRegistry and sets latched duration to zero.
+// Note: a <=0 latch means that the stats are reset on every call to Render().
+func DefaultStatsReceiver() StatsReceiver {
+	stat, _ := NewCustomStatsReceiver(nil, 0)
+	return stat
+}
```

Comment:
So one of the nice things in the Twitter Metrics lib is NullStatsReceiver, which is basically like a write to dev/null, nice for testing, because you end up passing stats receivers around in methods and constructors.  

Might be worth adding if this implementation requires additional setup or mocks to work correctly